packages/sol/src/verification.ts

```diff
@@ -0,0 +1,408 @@
+  if (tutorNotice <= MIN_TUTOR_NOTICE_DURATION) return FieldError.InvalidNotice;
+  if (tutorNotice > MAX_TUTOR_NOTICE_DURATION)
+    return FieldError.MaxNoticeExceeded;
+  return true;
+}
+
+export function isValidRuleTitle(
+  title: string
+): FieldError.TooShortRuleTitle | FieldError.TooLongRuleTitle | true {
+  if (title.length > MAX_RULE_TITLE_LENGTH) return FieldError.TooLongRuleTitle;
+  if (title.length < MIN_RULE_TITLE_LENGTH) return FieldError.TooShortRuleTitle;
+  return true;
+}
+
+export function isValidRuleStart(
+  ruleStart: string,
+  ruleEnd: string
+):
+  | FieldError.ISOInvalidRuleStart
+  | FieldError.ISOInvalidRuleEnd
+  | FieldError.RuleStartAfterEnd
+  | FieldError.RuleStartDatePassed
+  | true {
+  if (!Date.parse(ruleStart)) return FieldError.ISOInvalidRuleStart;
+  if (!Date.parse(ruleEnd)) return FieldError.ISOInvalidRuleEnd;
+  if (new Date(ruleStart) > new Date(ruleEnd))
+    return FieldError.RuleStartAfterEnd;
+  if (new Date() > new Date(ruleStart)) return FieldError.RuleStartDatePassed;
+  return true;
+}
+
+export function isValidRuleEnd(
+  ruleEnd: string,
+  ruleStart: string
+):
+  | FieldError.ISOInvalidRuleEnd
+  | FieldError.ISOInvalidRuleStart
+  | FieldError.RuleEndBeforeStart
+  | FieldError.RuleEndDatePassed
+  | true {
+  const currentTime = new Date().getTime();
+  const parsedRuleEnd = Date.parse(ruleEnd);
+  const parsedRuleStart = Date.parse(ruleStart);
+  const ruleEndTime = new Date(ruleEnd).getTime();
+  const ruleStartTime = new Date(ruleStart).getTime();
+
+  if (!parsedRuleEnd) return FieldError.ISOInvalidRuleEnd;
+  if (!parsedRuleStart) return FieldError.ISOInvalidRuleStart;
+  if (ruleEndTime < ruleStartTime) return FieldError.RuleEndBeforeStart;
+  if (ruleEndTime < currentTime) return FieldError.RuleEndDatePassed;
+  return true;
+}
+
+export function isValidRuleDuration(
+  duration: number
+): FieldError.InvalidRuleDuration | FieldError.MaxRuleDuratoinExceeded | true {
+  if (duration < MIN_RULE_DURATION_MINUTES)
+    return FieldError.InvalidRuleDuration;
+  if (duration > MAX_RULE_DURATION_MINUTES)
+    return FieldError.MaxRuleDuratoinExceeded;
+  return true;
+}
+
+export function isValidInterviewFeedback(
+  feedback: string
+):
+  | FieldError.TooLongInterviewFeedback
+  | FieldError.TooShortInterviewFeedback
+  | FieldError.InvalidInterviewFeedback
+  | true {
+  const feedbackText = getSafeInnerHtmlText(feedback);
+
+  if (!HTML_REGEX.test(feedback)) return FieldError.InvalidInterviewFeedback;
+  if (feedbackText.length < MIN_FEEDBACK_TEXT_LENGTH)
+    return FieldError.TooShortInterviewFeedback;
+  if (feedbackText.length > MAX_FEEDBACK_TEXT_LENGTH)
+    return FieldError.TooLongInterviewFeedback;
+  return true;
+}
+export function isValidInterviewNote(
+  note: string
+):
+  | FieldError.TooLongInterviewNote
+  | FieldError.TooShortInterviewNote
+  | FieldError.InvalidInterviewNote
+  | true {
+  const noteText = note.replace(HTML_TAGS_REGEX, "");
```

Comment:
Use `getSafeInnerHtml` here as well. 
```suggestion
const text = getSafeInnerHtml(note);
```